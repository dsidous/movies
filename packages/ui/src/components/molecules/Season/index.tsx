import React from 'react';
import { Link } from 'react-router-dom';
import { Typography } from '@material-ui/core';

import { MediaImage } from '@atoms/index';
import useStyles from './styles';
import { Season } from '@typesRoots/tv';

interface Props {
  season: Season;
  tvId: number;
}

const Season: React.FC<Props> = ({
  season: {
    poster_path,
    overview,
    name,
    episode_count,
    air_date,
    season_number,
  },
  tvId,
}) => {
  const classes = useStyles();

  const date = air_date !== null ? air_date.slice(0, 4) : '';
  const link = `/tv/${tvId}/season/${season_number}`;

  return (
    <div className={classes.root}>
      <Link to={link} className={classes.poster}>
        <MediaImage
          mediaType='poster'
          size={3}
          filePath={poster_path}
          name={name}
        />
      </Link>
      <div className={classes.details}>
        <Typography variant='h6'>{name}</Typography>
        <Typography
          paragraph
        >{`${date} | ${episode_count} episodes`}</Typography>
        <Typography variant='body2'>{overview}</Typography>
      </div>
    </div>
  );
};

export default Season;
