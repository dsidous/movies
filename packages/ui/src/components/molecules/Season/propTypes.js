import { object, number } from 'prop-types';

export const propTypes = {
  season: object.isRequired,
  tvId: number.isRequired,
};

export const defaultProps = {};
