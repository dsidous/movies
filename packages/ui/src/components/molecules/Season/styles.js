import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      display: 'grid',
      gridTemplateColumns: '1fr 4fr',
      gridColumnGap: 10,
      padding: '0 10px',
    },

    poster: {
      '& img': {
        cursor: 'pointer',
        width: '100%',
      },
    },

    details: {
      padding: 20,
    },

    last: {
      borderTop: theme.separator,
      margin: '40px auto 0',
      [theme.breakpoints.up('sm')]: {
        maxWidth: 960,
      },

      '& h4': {
        margin: '20px 10px',
      },
    },

    allBtn: {
      background: 'none',
      border: 'none',
      color: '#666',
      display: 'inline-block',
      fontSize: '1.2em',
      fontWeight: 'bold',
      margin: '20px 10px',
    },
  }),
  { name: 'season' },
);
