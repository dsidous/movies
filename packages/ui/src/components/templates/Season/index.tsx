import React from 'react';

import Episode from '../../molecules/Episode';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';
import useStyles from './styles';
import Skeleton from './skeleton';
import { Tv_Season } from '@typesRoots/tvseasons';

interface Props {
  loading: boolean;
  tvSeason: Tv_Season;
  tvId: string;
}

const Season: React.FC<Props> = ({
  loading,
  tvSeason: { name, air_date, episodes, poster_path },
  tvId,
}) => {
  const classes = useStyles();

  if (loading) {
    return <Skeleton />;
  }

  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={air_date}
        poster_path={poster_path}
        link={`/tv/${tvId}/seasons`}
        linkCopy='Back to seasons'
      />

      <ul className={classes.season__wrapper}>
        {episodes.map((episode) => (
          <li className={classes.season__element} key={episode.episode_number}>
            <Episode episode={episode} />
          </li>
        ))}
      </ul>
    </PageTransition>
  );
};

export default Season;
