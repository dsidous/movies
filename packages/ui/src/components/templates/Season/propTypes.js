import { object, number, bool } from 'prop-types';

export const propTypes = {
  loading: bool,
  tvSeason: object,
  tvId: number,
};

export const defaultProps = {};
