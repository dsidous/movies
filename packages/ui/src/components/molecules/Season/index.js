/* eslint-disable camelcase */
import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import MediaImage from '../../atoms/MediaImage';

const Season = props => {
  const {
    season: {
      poster_path,
      overview,
      name,
      episode_count,
      air_date,
      season_number,
    },
    tvId,
  } = props;

  const date = air_date !== null ? air_date.slice(0, 4) : '';
  const link = `/tv/${tvId}/season/${season_number}`;

  return (
    <div className="season">
      <Link to={link} className="season-poster">
        <MediaImage
          mediaType="poster"
          size={3}
          filePath={poster_path}
          name={name}
        />
      </Link>
      <div className="season-details">
        <h3>{name}</h3>
        <h5>{`${date} | ${episode_count} episodes`}</h5>
        <div>{overview}</div>
      </div>
    </div>
  );
};

Season.propTypes = propTypes;

export default Season;
