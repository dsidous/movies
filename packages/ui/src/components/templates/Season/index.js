/* eslint-disable camelcase */
import React from 'react';

import { propTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import Episode from '../../molecules/Episode';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';

const Season = props => {
  const { loading } = props;

  if (loading) {
    return <Spinner />;
  }

  const {
    tvSeason: { name, air_date, episodes, poster_path },
    tvId,
  } = props;

  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={air_date}
        poster_path={poster_path}
        link={`/tv/${tvId}/seasons`}
        linkCopy="Back to seasons"
      />

      <ul className="episodes__wrapper">
        {episodes.map(episode => (
          <li className="episodes__element" key={episode.episode_number}>
            <Episode episode={episode} />
          </li>
        ))}
      </ul>
    </PageTransition>
  );
};

Season.propTypes = propTypes;

export default Season;
