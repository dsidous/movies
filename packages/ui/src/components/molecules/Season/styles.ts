import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      [theme.breakpoints.up('sm')]: {
        display: 'grid',
        gridTemplateColumns: '1fr 4fr',
        gridColumnGap: theme.spacing(2),
      },
    },

    poster: {
      '& img': {
        cursor: 'pointer',
        width: '100%',
      },
    },

    details: {
      padding: theme.spacing(3),
    },
  }),
  { name: 'season' },
);
