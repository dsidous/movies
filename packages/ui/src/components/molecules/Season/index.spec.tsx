import React from 'react';
import { shallow } from 'enzyme';
import Season from '.';

const mockProps = {
  season: {
    poster_path: 'path/to/image',
    overview: 'overview',
    name: 'name',
    episode_count: 3,
    air_date: '01.01.2010',
    season_number: 2,
  },
  tvId: 1,
};

describe('Molecules/Season', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Season {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
