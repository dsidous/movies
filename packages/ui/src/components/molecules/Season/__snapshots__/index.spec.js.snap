// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Season should render as expected 1`] = `ShallowWrapper {}`;
